### ISSUE TITLE
bzr operations maybe don't take into account shared repos?

### ISSUE THREAD

**Issue by dpb1** [juju-solutions/bundletester#13]
When I test a branch from a shared repository, I get the following.  If I bzr export before testing, it works.  I'm not sure what value is being added that I'm missing, but it doesn't appear to take into account the shared repo when it copies the tree.

Perhaps the vcs operations should be done before the copy, or on the orig directory, and then use a `bzr export` to do the copy?

not sure.

```
Traceback (most recent call last):
  File "/home/dpb/src/canonical/bundletester/dpb1/.venv/bin/bundletester", line 9, in <module>
    load_entry_point('bundletester==0.5.0', 'console_scripts', 'bundletester')()
  File "/home/dpb/src/canonical/bundletester/dpb1/.venv/local/lib/python2.7/site-packages/bundletester/tester.py", line 89, in main
    suite = spec.SuiteFactory(options, options.testdir)
  File "/home/dpb/src/canonical/bundletester/dpb1/.venv/local/lib/python2.7/site-packages/bundletester/spec.py", line 248, in SuiteFactory
    model = classifier(directory, options)
  File "/home/dpb/src/canonical/bundletester/dpb1/.venv/local/lib/python2.7/site-packages/bundletester/spec.py", line 223, in CharmClassifier
    data = lp.infer_charm(directory) or {}
  File "/home/dpb/src/canonical/bundletester/dpb1/.venv/local/lib/python2.7/site-packages/bundletester/vcs.py", line 28, in infer_charm
    origin = self.get_origin(directory)
  File "/home/dpb/src/canonical/bundletester/dpb1/.venv/local/lib/python2.7/site-packages/bundletester/vcs.py", line 12, in get_origin
    b = branch.Branch.open(directory)
  File "/home/dpb/src/canonical/bundletester/dpb1/.venv/local/lib/python2.7/site-packages/bzrlib/branch.py", line 188, in open
    possible_transports=possible_transports)
  File "/home/dpb/src/canonical/bundletester/dpb1/.venv/local/lib/python2.7/site-packages/bzrlib/bzrdir.py", line 1079, in open_branch
    possible_transports=possible_transports)
  File "/home/dpb/src/canonical/bundletester/dpb1/.venv/local/lib/python2.7/site-packages/bzrlib/branch.py", line 2028, in open
    found_repository = a_bzrdir.find_repository()
  File "/home/dpb/src/canonical/bundletester/dpb1/.venv/local/lib/python2.7/site-packages/bzrlib/bzrdir.py", line 571, in find_repository
    raise errors.NoRepositoryPresent(self)
bzrlib.errors.NoRepositoryPresent: No repository present: "file:///tmp/bundletester-QmSb_p/backends-support/"
```

**Comment by stub42** [comment]
I see this too. I'd suggest dropping this whole code path for local directories, and just using shutil.copytree if a copy is actually needed at all and avoiding 'only committed files' issues.

aaron or john could both tell you the correct invocation in their sleep, so maybe ping them.

**Comment by tvansteenburgh** [comment]
This is the same as #12. I'm going to close this one and fix it under #12. Thanks for the report.
